src/api.rs

```diff
@@ -1,30 +1,36 @@
 use std::str::FromStr;
 
-use bitcoin::secp256k1::Scalar;
 use flutter_rust_bridge::StreamSink;
 
 use crate::{
-    constants::{LogEntry, ScanProgress, WalletStatus, WalletType},
-    db::{self},
+    constants::{LogEntry, WalletType},
     electrumclient::create_electrum_client,
     nakamotoclient,
-    spclient::{self, get_sp_client, derive_keys_from_mnemonic},
+    spclient::{ScanProgress, SpClient, derive_keys_from_mnemonic},
     stream::{self, loginfo},
 };
 
 const PASSPHRASE: &str = ""; // no passphrase for now
 
+pub struct WalletStatus {
+    pub amount: u64,
+    pub birthday: u32,
+    pub scan_height: u32,
+    pub tip_height: u32,
+}
+
 pub fn create_log_stream(s: StreamSink<LogEntry>) {
     stream::create_log_stream(s);
 }
-pub fn create_amount_stream(s: StreamSink<u32>) {
+pub fn create_amount_stream(s: StreamSink<u64>) {
     stream::create_amount_stream(s);
 }
 pub fn create_scan_progress_stream(s: StreamSink<ScanProgress>) {
     stream::create_scan_progress_stream(s);
 }
 
 pub fn setup(
```

Comment:
I think the `setup` function has now changed it's purpose: you're only meant to call it once when creating a new wallet. Maybe it should be renamed to`create_new_wallet` or something?

Reply:
I don't like create_new_wallet because actually it can be a restoration too, idk

Reply:
Actually, the `setup` function also calls `nakamotoclient::setup`, so it's not just creating a new wallet, my bad

Reply:
Yes but I'm currently moving nakamotoclient::setup out of it, so you still have a point. I feel that `setup_wallet` is still less misleading since we're not necessarily creating a wallet, but maybe I'll change my mind